x/campaign/types/campaign_test.go

```diff
@@ -45,6 +45,17 @@ func TestCampaign_Validate(t *testing.T) {
 	))
 	require.True(t, campaign.IsTotalSharesReached(totalSharesReached.AllocatedShares, spntypes.TotalShareNumber))
 
+	invalidSpecialAllocations := campaign.NewSpecialAllocations(
+		sample.Shares(r),
+		campaign.Shares(sdk.NewCoins(
+			sdk.NewCoin("foo", sdk.NewInt(100)),
+			sdk.NewCoin("s/bar", sdk.NewInt(200)),
+		)),
+	)
+	require.Error(t, invalidSpecialAllocations.Validate())
+	campaignInvalidSpecialAllocations := sample.Campaign(r, 0)
+	campaignInvalidSpecialAllocations.SpecialAllocations = invalidSpecialAllocations
+
```

Comment:
I think we should add a test where we are validating a "non-empty" special allocations field in the `campaign` object.

The best way to do this is to probably modify `sample.Campaign()` to automatically make that field non-null but valid